app/models/folio/item_with_status.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Folio
+  # Other status that we aren't using include "Unavailable", "On order", "In process", "Intellectual item", "In transit"
+  STATUS_CHECKED_OUT = 'Checked out'
+  STATUS_AVAILABLE = 'Available'
+  STATUS_HOLD = 'Awaiting pickup'
+
+  PAGE_LOCATIONS = %w(HILA-SAL3-STACKS
+                      HILA-SAL3-STACKS
+                      SPEC-SAL3-FELTON
+                      SPEC-SAL3-GUNST
+                      SPEC-SAL3-MEDIA
+                      SPEC-SAL3-MSS
+                      SPEC-SAL3-RBC
+                      SPEC-SAL3-U-ARCHIVES
+                      SPEC-SAL3-U-ARCHIVES).freeze
+
+  ItemWithStatus = Data.define(:barcode, :status, :request_status, :type, :public_note, :permanent_location, :temporary_location,
+                               :callnumber) do
+    def checked_out?
+      status == STATUS_CHECKED_OUT
+    end
+
+    # TODO, is this complete?
+    def status_class
+      status == STATUS_AVAILABLE ? 'available' : 'unavailable'
+    end
+
+    # TODO, we probably need to handle "Page", which is something Symphony had.
+    def status_text
+      status
+    end
+
+    # TODO, HUH?
+    def current_location_code
```

Comment:
To be removed once https://github.com/sul-dlss/sul-requests/pull/1501 is merged and this PR is rebased.